packages/superset-ui-chart-companion/test/legend/WithLegend.test.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable import/first */
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+
+jest.mock('resize-observer-polyfill');
+// @ts-ignore
+import { triggerResizeObserver } from 'resize-observer-polyfill';
+import { WithLegend } from '../../src';
+
+let renderChart = jest.fn();
+let renderLegend = jest.fn();
+
+describe('WithLegend', () => {
+  beforeEach(() => {
+    renderChart = jest.fn(() => <div className="chart" />);
+    renderLegend = jest.fn(() => <div className="legend" />);
+  });
+
+  it('sets className', () => {
+    const wrapper = shallow(
+      <WithLegend className="test-class" renderChart={renderChart} renderLegend={renderLegend} />,
+    );
+    expect(wrapper.hasClass('test-class')).toEqual(true);
+  });
+
+  it('renders when renderLegend is not set', done => {
+    const wrapper = mount(<WithLegend width={500} height={500} renderChart={renderChart} />);
+
+    triggerResizeObserver();
+    // Have to delay > 300ms
+    // because the default debounce time for ParentSize is 300.
+    setTimeout(() => {
+      expect(renderChart).toHaveBeenCalledTimes(1);
+      expect(wrapper.render().find('div.chart')).toHaveLength(1);
+      expect(wrapper.render().find('div.legend')).toHaveLength(0);
+      done();
+    }, 500);
+  });
+
+  it('renders', done => {
+    const wrapper = mount(
+      <WithLegend width={500} height={500} renderChart={renderChart} renderLegend={renderLegend} />,
+    );
+
+    triggerResizeObserver();
+    // Have to delay > 300ms
+    // because the default debounce time for ParentSize is 300.
+    setTimeout(() => {
+      expect(renderChart).toHaveBeenCalledTimes(1);
+      expect(renderLegend).toHaveBeenCalledTimes(1);
+      expect(wrapper.render().find('div.chart')).toHaveLength(1);
+      expect(wrapper.render().find('div.legend')).toHaveLength(1);
+      done();
+    }, 500);
+  });
+
+  it('renders without width or height', done => {
+    const wrapper = mount(<WithLegend renderChart={renderChart} renderLegend={renderLegend} />);
+
+    triggerResizeObserver();
+    // Have to delay > 300ms
+    // because the default debounce time for ParentSize is 300.
```

Comment:
you could expose this prop so this is more declarative.

Reply:
great comment!